src/devices/Bmp180/Bmp180.cs

```diff
@@ -0,0 +1,304 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+//Ported from https://github.com/adafruit/Adafruit_Python_BMP/blob/master/Adafruit_BMP/BMP085.py  
+//Formulas and code examples can also be found in the datasheet https://cdn-shop.adafruit.com/datasheets/BST-BMP180-DS000-09.pdf
+
+using System;
+using System.Device.I2c;
+using System.Buffers.Binary;
+using System.Threading;
+
+namespace Iot.Device.Bmp180
+{
+    public class Bmp180:IDisposable
+    {
+        private I2cDevice _i2cDevice;
+        private readonly CommunicationProtocol _communicationProtocol;
+        private bool _initialized = false;
+        private readonly CalibrationData _calibrationData;
+        private Sampling _mode = Sampling.Standard;
+
+        private enum CommunicationProtocol
+        {
+            I2c
+        }
+
+        public Bmp180(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice;            
+            _calibrationData = new CalibrationData();
+            _communicationProtocol = CommunicationProtocol.I2c;
+        }
+
+        private void Begin()
+        {
+            _initialized = true;
+
+            //Read the coefficients table
+            _calibrationData.ReadFromDevice(this);
+        }
+
+        /// <summary>
+        /// Sets sampling to the given value
+        /// </summary>
+        /// <param name="sampling"></param>
+        public void SetSampling(Sampling mode)
+        {
+            this._mode = mode;
+        }
+
+        /// <summary>
+        ///  Reads the temperature from the sensor
+        /// </summary>
+        /// <returns>
+        ///  Temperature in degrees celsius
+        /// </returns>
+        public double ReadTemperature()
+        {
+            // Make sure the I2C device is initialized
+            if (!_initialized)
+            {
+                Begin();
+            }
+            //Gets the compensated temperature in degrees celsius
+            var UT = ReadRawTemperature();

```

Comment:
We avoid using `var` so that the code is more readable. The only time where we allow it is whenever it belongs to a new statement like: `var str = new String();` since in that case you absolutely know what the type of str is.